hypothesis-python/tests/cover/test_core.py

```diff
@@ -113,3 +115,16 @@ def test_can_find_with_db_eq_none():
 def test_no_such_example():
     with pytest.raises(NoSuchExample):
         find(s.none(), bool, database_key=b"no such example")
+
+
+def test_validates_strategies_for_test_method():
+    invalid_strategy = s.lists(s.nothing(), min_size=1)
+
+    class TestStrategyValidation(TestCase):
+        @given(invalid_strategy)
+        def test_method_with_bad_strategy(self, x):
+            pass
+
+    instance = TestStrategyValidation("test_method_with_bad_strategy")
+    with pytest.raises(InvalidArgument):
+        instance.test_method_with_bad_strategy()
```

Comment:
I don't know enough about `TestCase` to know whether this is the proper way to manually invoke a test method.

Reply:
Actually, it occurs to me that I can just as easily test this without involving `TestCase` at all, and simply extend `object` instead.